build_scripts/build-script-files/jxcore/build_scripts/android_compile.sh

```diff
@@ -0,0 +1,222 @@
+#!/bin/bash
+
+# Copyright & License details are available under JXCORE_LICENSE file
+
+NORMAL_COLOR='\033[0m'
+RED_COLOR='\033[0;31m'
+GREEN_COLOR='\033[0;32m'
+GRAY_COLOR='\033[0;37m'
+
+LOG() {
+  COLOR="$1"
+  TEXT="$2"
+  echo -e "${COLOR}$TEXT ${NORMAL_COLOR}"
+}
+
+
+ERROR_ABORT() {
+  if [[ $? != 0 ]]
+  then
+    LOG $RED_COLOR "compilation aborted\n"
+    exit  
+  fi
+}
+
+
+ERROR_ABORT_MOVE() {
+  if [[ $? != 0 ]]
+  then
+    $($1)
+    LOG $RED_COLOR "compilation aborted for $2 target\n"
+    exit  
+  fi
+}
+
+if [ $# -eq 0 ]
+then
+  LOG $RED_COLOR "no argument provided."
+  LOG $GREEN_COLOR "usage: android_compile <ndk_path> <optionally --embed-leveldown>\n"
+  exit
+fi
+
+export ANDROID_NDK=$1
+
+CONF_EXTRAS=
+MIPS=0 #out_mipsel_droid
+ARM64=0 #out_arm64_droid
+
+if [ $# -eq 2 ]
+then
+  CONF_EXTRAS=$2
+fi
+
+ARM7=out_arm_droid
+INTEL64=out_x64_droid
+INTEL32=out_ia32_droid
+FATBIN=out_android/android
+    
+MAKE_INSTALL() {
+  TARGET_DIR="out_$1_droid"
+  mv $TARGET_DIR out
+  ./configure --static-library --dest-os=android --dest-cpu=$1 --engine-mozilla --compress-internals $CONF_EXTRAS
+  ERROR_ABORT_MOVE "mv out $TARGET_DIR" $1
+  make V= -j 8
+  ERROR_ABORT_MOVE "mv out $TARGET_DIR" $1
+  
+  PREFIX_DIR="out/Release"
+  $STRIP -d $PREFIX_DIR/libcares.a
```

Comment:
I didn't write that code and I didn't spend time to review it beside speeding up the compilation ;)